### PR TITLE
added mean bias loss function

### PR DESCRIPTION
**Pull request by codeboy5**
References: 
> https://www.sciencedirect.com/topics/engineering/mean-bias-error
> https://medium.com/human-in-a-machine-world/mae-and-rmse-which-metric-is-better-e60ac3bde13d

### CODE REVIEW

**Comment by kartikdutt18** [comment]
Hi @codeboy5, Could you add a description to the PR that include a source of reference. Thanks.

File: src/mlpack/methods/ann/loss_functions/mean_bias_error.hpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The mean bias error performance function measures the network's
+ * performance according to the mean of errors.
+ *
+ * @tparam ActivationFunction Activation function used for the embedding layer.
```
Comment (kartikdutt18) [inline src/mlpack/methods/ann/loss_functions/mean_bias_error.hpp]: I don't think this is required.

File: src/mlpack/methods/ann/loss_functions/mean_bias_error.hpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+  /**
+   * Computes the mean bias error function.
+   *
+   * @param input Input data used for evaluating the specified function.
+   * @param target The target vector.
+   */
+  template<typename InputType, typename TargetType>
+  double Forward(const InputType&& input, const TargetType&& target);
+  /**
```
Comment (kartikdutt18) [inline src/mlpack/methods/ann/loss_functions/mean_bias_error.hpp]: Could you leave an empty line here. Thanks.

File: src/mlpack/methods/ann/loss_functions/mean_bias_error.hpp [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                const TargetType&& target,
+                OutputType&& output);
+
+  //! Get the output parameter.
+  OutputDataType& OutputParameter() const { return outputParameter; }
+  //! Modify the output parameter.
+  OutputDataType& OutputParameter() { return outputParameter; }
+
+  /**
+   * Serialize the layer
```
Comment (kartikdutt18) [inline src/mlpack/methods/ann/loss_functions/mean_bias_error.hpp]: Could you add a full stop here.

File: src/mlpack/methods/ann/loss_functions/mean_bias_error_impl.hpp [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename TargetType, typename OutputType>
+void MeanBiasError<InputDataType, OutputDataType>::Backward(
+    const InputType&& input,
+    const TargetType&& target,
+    OutputType&& output)
+{
+  output = -1;
```
Comment (kartikdutt18) [inline src/mlpack/methods/ann/loss_functions/mean_bias_error_impl.hpp]: Hi @codeboy5, This is will make output = -1 , a single value however we would want it to have same size as input. You can try output.set_size(arma::size(input)) and use fill to fill it with -1, or any other way you feel comfortable with.

**Comment by codeboy5** [comment]
> Hi @codeboy5, Could you add a description to the PR that include a source of reference. Thanks.

Sure, will do that

File: src/mlpack/methods/ann/loss_functions/mean_bias_error_impl.hpp (hunk 4 above)
Comment (codeboy5) [reply src/mlpack/methods/ann/loss_functions/mean_bias_error_impl.hpp]: how do use we fill ?

Comment (kartikdutt18) [reply src/mlpack/methods/ann/loss_functions/mean_bias_error_impl.hpp]:
```
output.set_size(arma::size(input));
output.fill(-1.0);
```

**Comment by codeboy5** [comment]
> Hi @codeboy5, Could you add a description to the PR that include a source of reference. Thanks.

Hey, I saw references to this error function in : 
>https://www.sciencedirect.com/topics/engineering/mean-bias-error
>https://medium.com/human-in-a-machine-world/mae-and-rmse-which-metric-is-better-e60ac3bde13d

Should I just add these links ?

**Comment by kartikdutt18** [comment]
Add them in PR, could you also find a paper for the same or any library that uses them.

File: src/mlpack/methods/ann/loss_functions/mean_bias_error_impl.hpp [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename TargetType>
+double MeanBiasError<InputDataType, OutputDataType>::Forward(
+    const InputType&& input, const TargetType&& target)
+{
+  return arma::accu(target-input) / target.n_cols;
```
Comment (zoq) [inline src/mlpack/methods/ann/loss_functions/mean_bias_error_impl.hpp]: Pedantic style comment, missing space between the `-` operation.

**Comment by codeboy5** [comment]
Hi, @kartikdutt18 is this error related to my recent changes?

**Comment by birm** [comment]
@codeboy5 the macOS Julia error has nothing to do with your code or anything in mlpack; it's a [checksum mismatch](https://dev.azure.com/mlpack/mlpack/_build/results?buildId=602&view=logs&j=7fdb82b0-4024-50f2-cd8f-419b00dd5cf6&t=48d602b1-1a3a-57b2-6228-666bf2ff2980&l=163).
As for the other failure that has just come (mlpack.mlpack (Windows VS16 Plain)), it's related to FastLSTMDistractedSequenceRecallTest, and I've seen this come up occasionally. It has nothing to do with this PR.
You can check an azure pipeline build output by clicking "Details" next to it, and from there clicking "View more details on Azure Pipelines"

**Pull request merged**